[16.0][MIG] account_payment_order: Migration to 16.0

wpichler opened the pull request:
https://github.com/OCA/bank-payment/issues/964

Needs https://github.com/OCA/bank-payment/pull/969

Standard Migration - just one thing - it was necessary to add groups="base.group_multi_company" to the order_id field in account_payment_line_form. Without this i do get this error:

Field 'company_id' used in domain of field 'order_id' ([('company_id', 'in', [company_id, False])]) is restricted to the group(s) base.group_multi_company.


order_id is check_company - so company_id domain gets added indirectly and should be ok without this group. Maybe because of the related company_id this problem occours ?

alexis-via committed:
Start to port bank-payment to v9 (with a lot of improvements) during the Sorrento Code sprint 2016

Improvements include:
- full re-organisation of modules and big re-organisation of the code
- simplification of the code related to the fact that support for direct debit is now in t
he base module, not added by an optional module account_direct_debit (module was removed)
- new design of the wizard to select move lines to pay
- support for non-SEPA file transfer-
- support for German direct debit SEPA files (fixes bug #129)
- remove workflow of payment.order
- add wizard of selection of move lines to pay

sbidoul committed:
[FIX+IMP] account_payment_order: 2 things:

* IMP: prevent deleting a move-line that is linked to a payment order line
* FIX: better filtering of move lines already included in open payment order

alexis-via committed:
Add support for multi-currency in the generation of the account move.

Take into account all the remaining remarks of Frederic Clementi dated May 30:
- see multi-currency amounts in the wizard to add transaction lines
- add help msg on bank payment lines

pedrobaeza committed:
[IMP] account_payment_order: Remove useless migration script

yvaucher committed:
Move the account_payment_mode ir.rule in account_payment_mode module

sbidoul committed:
[FIX+IMP] account_payment_order: 2 more things:

* IMP: extensibility of _prepare_move

  Pass the list of bank lines to _prepare_move so it is possible to customize the move (eg it's name) based on the lines being paid.

* FIX: payment order: add missing onchange dependencies in move selection wizard

alexis-via committed:
[FIX] account_payment_order: 4 things:

* Fix crash when selecting a customer that has a direct debit payment mode but no mandate
* Hide many fields when payment_order_ok is False on a payment_mode
* Fix payment line communication when the move line is not linked to an invoice (problem found when porting donation_debit_debit to v9)
* For the communication field, if self.move_id.ref is False, fallback on self.move_id.name

sbidoul committed:
[IMP] set partner on offsetting account move line if possible

fixes #277

yvaucher committed:
Reactivate company rules in account_payment_order

Commit:
Port almost all modules to v10 (#305)

Port almost all modules to v10

* Update to EPC Rulebook v9.2 that start to apply on 2016-11-20 (bug #300)

Commit:
[FIX] account_payment_order: Use 'index' instead of 'select'

jalzaga committed:
Remove internal_type=other restriction for transfer account (#311)

alexis-via committed:
Creation/Modification of bank info only for Account Payment group

vrenaville committed:
[FIX] fill date instead off maturity date on account move line generated

mdietrichc2c committed:
Fix 2 issues on form view for payment order

alexis-via committed:
[FIX+IMP] account_payment_order:

* FIX: Bug #353 crash when selecting a payment mode that has a variable link to a bank account
* FIX: Add 'post_move' option on account.payment.mode with default True, to keep the previous behavior
* IMP: Explicit error when bank account is missing on bank journal
* IMP: Add field default_date_prefered on payment mode.

GillesTephaneMeyomesse committed:
[ADD] description field for payment order

mdietrichc2c committed:
[IMP] account_payment_order: Don't allow deletion of an uploaded payment order + allow to cancel an uploaded payment order

grindtildeath committed:
[IMP] account_payment_order: Restriction + tests

* Restrict the deletion of bank payment line
* Add test_cancel_payment_order

pedrobaeza committed:
[FIX] account_payment_order: Adapt test to v10

Commit:
[IMP] account_payment_order: select partner bank account automatically.

grindtildeath committed:
[10.0]Fix payment method onchange with multicompany (#374)

GillesTephaneMeyomesse committed:
[ADD] report for payment order (#370)

etobella committed:
[MIG] account_payment_order

JordiBForgeFlow committed:
When the user looks for open payables or receivables, in the
context of payment orders, she should ocus on the amount that
is due to be paid. In this method we are forcing to display both
the amount due in company and in the invoice currency.

We then hide the fields debit and credit, because they add no value.

pedrobaeza committed:
[FIX] account_payment_order: Correct domain definition on account.payment.line view

astirpe committed:
[11.0][FIX] account_payment_order: printing report

pedrobaeza committed:
[FIX] account_payment_order: Take payment mode from move (#466)

When adding to a payment order from invoices, the payment mode
considered is the one in the invoice, but once you have confirmed it,
you can't change it, so if you change the payment mode in the journal
items, it doesn't make the difference.

With this change, this is taken into account for adding to the
payment order(s) properly.

carlosdauden committed:
[FIX] account_payment_order: Add bank payment line multi-company rule

alexis-via committed:
Several small usability improvements

- when manually encoding a payment line, get company currency as default currency
- Search on payment order numbers
- Direct search on bank journal name in payment orders
- remove widget="selection" on account.payement.mode : this object is rarely modified, so widget='selection' is not a time saver, but we miss the HTML link, which can be pretty convenient to check the configuration.

etobella committed:
[FIX] report should show the invoice number, not the move name (#477)

[IMP] account_payment_order: report show the invoice number or the move name

pedrobaeza committed:
[FIX] account_payment_order: Fix FR translation

etobella committed:
Translated using Weblate (Español (España))

Currently translated at 100,0% (193 of 193 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_payment_order/es/

etobella committed:
[FIX] default_partner_id added on partner_bank if it is created.

Commit:
[10.0] Add filter on partners on wizard "Create Payment Lines from Journal Items" (#489)

* Add filter on partners on wizard "Create Payment Lines from Journal Items"

* Bump version number

ecino committed:
[IMP] account_payment_order: Use bank.payment.line reference in Payment Order Move Lines (#418)

Martronic-SA committed:
Fix structured communication

Adds 'structured' to the invoice_reference_type2communication_type ans the 'structured' type is added by the module itself in account_invoice but breaks the account_move_line.py as explained here: https://github.com/OCA/bank-payment/issues/495

mncoon committed:
Translated using Weblate (German)

Currently translated at 50.8% (98 of 193 strings)

Translation: bank-payment-11.0/bank-payment-11.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-11-0/bank-payment-11-0-account_payment_order/de/

sergio-teruel committed:
[12.0][mig] account_payment_order: Migrate to v12.0

pedrobaeza committed:
[MIG] account_payment_order: Finish migration to v12

* Removed deprecated YAML tests.
* Removed document dependency, as now there's no attachment dropdown.
* Switched deprecated `compute` method of res.currency.
* Restored missing field `reference_type` in invoices.
* Fixed call to `message_post`.
* Added missing description in model.

pedrobaeza committed:
[MIG] account_banking_sepa_direct_debit: Finish migration to 12.0

* Fixed tests
* Adjust code for Python dates handling
* Update copyrights

ThomasBinsfeld committed:
[REF] Account Payment Order: add Admin in security group

ThomasBinsfeld committed:
[REF] Payment Order: better performance of draft2open method
Add a norecompute in draft2open to avoid the recomputation of total_company_currency n times (where n is the number of payment lines)

Commit:
Translated using Weblate (German)

Currently translated at 44.9% (97 of 216 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/de/

rven committed:
Translated using Weblate (Dutch)

Currently translated at 62.0% (134 of 216 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/nl/

pedrobaeza committed:
[FIX] account_payment_order: Fix journal domain

Steps to reproduce
------------------

* Create a payment mode with "Link to Bank Account" = "Variable".
* Select several journals.
* Create a payment order with that payment mode.
* Save.
* Go to other menu and come back to payment order.
* Try to select a journal: the list is empty.

Fixes #544

ThomasBinsfeld committed:
[REF] Payment Order: allow add to order from invoice if order exists

ThomasBinsfeld committed:
[REF] Payment Order: payment line creation from move line
Benefits of the batch creation of Odoo 12.0
The method returns the created payment lines

ThomasBinsfeld committed:
[REF] Account Payment Order: split generate_move method

ThomasBinsfeld committed:
[ADD] Payment Order: only payment order option on payment methods
New option on payment methods: payment order only (unchecked by default) allowing to enforce the use of payment orders for some payment methods

Commit:
Translated using Weblate (Italian)

Currently translated at 5.5% (12 of 217 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/it/

Commit:
[FIX] account_payment_order: wrong function name

Commit:
[FIX] account_payment_order: report totals

Commit:
[FIX] account_payment_order: report title

ThomasBinsfeld committed:
[REF] Payment Order: move all move preparation code in _prepare_move

rven committed:
[12.0] [FIX] Failing test test_account_payment_02

Commit:
Add indexes on account payment models

The fields where the indexes are added are used in searches in
account_payment_order, which becomes really slow when a database have
many lines.

pedrobaeza committed:
[FIX] account_payment_error: Avoid singleton error in allowed_journal_ids

Fixes #612

pedrobaeza committed:
[FIX+IMP] account_payment_order: Make tests resilient + inbound inheritable + SavepointCase

Previous tests did several risk operations on tests, like performing an unbound search
on payment orders, what can lead to errors if there are some data on DB (or for example
we add in the future some payment orders in demo.

This commit also split the tests for inbound payment in a base inheritable class that
can be reused in other dependant modules, and the test of this module.

Finally, we use SavepointCase for improving performance in tests.

carlosdauden committed:
[IMP] account_payment_order: Select first partner bank

Commit:
Translated using Weblate (Catalan)

Currently translated at 100.0% (223 of 223 strings)

Translation: bank-payment-12.0/bank-payment-12.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-12-0/bank-payment-12-0-account_payment_order/ca/

sergio-teruel committed:
[FIX] account_payment_order: Fix payment order report. Adapt to bootstrap4

pedrobaeza committed:
[FIX] account_payment_order: Sequences must be noupdate=1

pedrobaeza committed:
[FIX] account_payment_order: Restore colors in payment order tree view

pedrobaeza committed:
[IMP] account_payment_order: Print payment order in user lang if not generated

When you haven't still generated the file, `generated_user_id` is not filled, and
so, you can't control the language in which the payment order is going to be printed.

Adding a condition for using the user language in that case makes this more
comfortable.

Commit:
[IMP] account_payment_order: change 'bill reference' column value in reports

CarlesAntoli committed:
Added translation using Weblate (Catalan)

rven committed:
[IMP] account_payment_order: black, isort

rven committed:
[MIG] account_payment_order: Migration to 13.0

astirpe committed:
[FIX] multicompany rule

ValentinVinagre committed:
[FIX] account_payment_order: fix set partner_bank_id in account move lines
Case:
The bank account isn't set properly in the payment order lines, because the bank account is only got from account.move.line or the first bank account in the partner_id, but never from the invoice (account.move).

Solution:
With this fix, the bank account will be get from account.move when matching the criteria.

StefanRijnhart committed:
[FIX] Set priority on alternate view

ValentinVinagre committed:
[FIX] account_payment_order: fix payment orders from invoice tree view
Case: User can't add invoices to payment/debit orders from the tree view but it's still possible to add them one by one from the form view.
- Remove binding_views="form": by default are "tree & form".
With this fix, users can add payment/debit orders from the invoice tree view

ValentinVinagre committed:
[FIX] account_payment_order: fix the attachment file name & dowload.
Case: After the configuration of a debit order and generating the .xml file, user can't download it from the next step screen using the download link.
This fix makes the .xml file downloadable from the .xml file name as it was in older versions

alexis-via committed:
Fix state condition on invoice view

luc-demeyer committed:
[13.0][IMP]move payment order menus

luc-demeyer committed:
move Debit Order after Payments in Customer menu

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (220 of 220 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_payment_order/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (220 of 220 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_payment_order/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (220 of 220 strings)

Translation: bank-payment-13.0/bank-payment-13.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-13-0/bank-payment-13-0-account_payment_order/es_AR/

astirpe committed:
[FIX] avoid cache miss in compute

alan196 committed:
[MIG] account_payment_order: Migration to 14.0

CarlosRoca13 committed:
[MIG] account_payment_order: Migration to v14

oca-travis committed:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[UPD] README.rst

joao-p-marques committed:
[FIX] account_payment_order: Fix tests

On multi-company settings, we need to make sure we are searching and creating objects with the correct company associated.

TT28423

OCA-git-bot committed:
account_payment_order 14.0.1.0.1

joao-p-marques committed:
[FIX] account_payment_order: Fix tests

Don't use taxes in invoices to avoid inconsistency across localization settings
Create account instead of relying on environment

OCA-git-bot committed:
account_payment_order 14.0.1.0.2

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 86.8% (192 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/fr_FR/

joao-p-marques committed:
[FIX] account_payment_order: Fix tests

After https://github.com/odoo/odoo/commit/fcaa54939e9a4f0dd5e47cd0ccffe7aa24bd451c, we need to specify the invoice date for vendor bills to be able to post it

OCA-git-bot committed:
account_payment_order 14.0.1.0.3

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 91.4% (202 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/fr_FR/

pedrobaeza committed:
[FIX] account_payment_order: Computed non-stored fields needs always a value

We should assign a value for all records of the set.

Commit:
Added translation using Weblate (Spanish (Chile))

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 49.3% (109 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es_CL/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (221 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 100.0% (221 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es_CL/

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 100.0% (221 of 221 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es_CL/

baimont committed:
[13.0][IMP] account_payment_order - move date should be bank line date when offsetting_account is bank_account

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 68.1% (150 of 220 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/nl/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 70.4% (155 of 220 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/nl/

pedrobaeza committed:
[FIX] account_payment_order: Remove old migration scripts

OCA-git-bot committed:
account_payment_order 14.0.1.0.5

joao-p-marques committed:
[FIX] account_payment_order: Change security group category

Before this change, the group was creating a new section in each user's settings called "Other" with a field called "Other Extra Rights" that could select "Accounting / Payments" as an option
This places it in the correct location

TT29182

joao-p-marques committed:
[FIX] account_payment_order: Make post move field invisible in payment modes

When the generate_moves option is not selected, the post_move option should not be available

TT29182

joao-p-marques committed:
[FIX] account_payment_order: fix account move views

Current spec was causing weird display in account move view, both for customer invoices and vendor bills

Changing the placement and adding back the label fixes the issue

TT29027

OCA-git-bot committed:
account_payment_order 14.0.1.1.0

OCA-git-bot committed:
account_payment_order 14.0.1.2.0

manuelregidor committed:
[IMP]account_payment_order: payment date when adding transaction line

alexis-via committed:
Improve computation of sepa on account.payment.order: check IBAN is in SEPA zone

Update move line generation to get transfer account from bank journal
Update payment mode configuration accordingly (3 fields removed)
Several improvements in payment order tree and form view

alexis-via committed:
account_payment_order: remove done state (migration script provided)

Don't set reference_type field on account.move required=True, because it
causes a lot of problems in other modules that don't depend on this one.
Setting it as required in the view is enough.

alexis-via committed:
account_payment_partner: add optional="hide" on payment mode in invoice tree view

AmmarOfficewalaSerpentCS committed:
[IMP] account_payment_order: Add hook method for finding previous payment orders

OCA-git-bot committed:
account_payment_order 14.0.1.2.1

AmmarOfficewalaSerpentCS committed:
[IMP] account_payment_order: Pass single move in a get_account_payment_domain mehtod to find move in a method.

OCA-git-bot committed:
account_payment_order 14.0.1.2.2

OCA-git-bot committed:
account_payment_order 14.0.1.2.3

alexis-via committed:
FIX crash when communication=null on payment line

alexis-via committed:
account_payment_order: payment_mode_id readonly when state != draft

iammiracle committed:
[FIX] account_payment_order: Bind report to model

joao-p-marques committed:
[FIX] account_payment_order: Fix multi-company issues in tests

Use base `account` tests from Odoo to avoid issues with the environment.

OCA-git-bot committed:
account_payment_order 14.0.1.2.4

alexis-via committed:
Update test suite

Fix bug in account_banking_mandate when a res.partner.bank is not attached to a particular company (company_id is NOT required on res.partner.bank)

oca-travis committed:
[UPD] Update account_payment_order.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (212 of 212 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/es_AR/

simonstromb committed:
Added translation using Weblate (Swedish)

simonstromb committed:
Translated using Weblate (Swedish)

Currently translated at 56.6% (120 of 212 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/sv/

simonstromb committed:
Translated using Weblate (Swedish)

Currently translated at 63.6% (135 of 212 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/sv/

simonstromb committed:
Translated using Weblate (Swedish)

Currently translated at 65.0% (138 of 212 strings)

Translation: bank-payment-14.0/bank-payment-14.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-14-0/bank-payment-14-0-account_payment_order/sv/

OCA-git-bot committed:
account_payment_order 14.0.1.4.0

misern2 committed:
[MIG] account_payment_order: Migration to 15.0

alexis-via committed:
account_payment_order: communication required in form view

MiquelRForgeFlow committed:
[IMP] account_payment_order: don't requiere reference_type anymore

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
[UPD] README.rst

juliap98 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (212 of 212 strings)

Translation: bank-payment-15.0/bank-payment-15.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-15-0/bank-payment-15-0-account_payment_order/ca/

rousseldenis committed:
[14.0][IMP] account_payment_order: Add existing payment references to communication

If some movements have been reconciled with the original invoice,
their references should be added in communication too.

e.g.: Manual credit notes

etobella committed:
[IMP] account_payment_order: Promote to Mature

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_payment_order 15.0.1.0.1

rousseldenis committed:
[14.0][FIX] account_payment_order: Don't duplicate communication reference

pedrobaeza committed:
[FIX] account_payment_order: Don't fail due to tz

When the test hour is around midnight, and demo data using CET/CEST
timezones, current code fails due to some asserts not having this into
consideration. This fix avoids such problem, and irons the execution
no matter the hour.

ramiadavid committed:
[FIX] account_payment_order: Fix tests

Some tests need to be run on post-install
The bank_journal.inbound_payment_order_only field is true only if all inbound_payment_method_ids are payment_order_only True. For this we must include the "SEPA Direct Debit for customers" in test

OCA-git-bot committed:
account_payment_order 15.0.1.0.3

alexis-via committed:
[FIX] account_payment_order: fix action returned by create_account_payment_line()

create_account_payment_line() is supposed to returned a form view for a
single payment order and a tree view for multiple payment orders. Before
this fix, it would return a tree view if you were payment more than 1
invoices while generating a single payment order, because the list
result_payorder_ids would contain several time the same ID.

OCA-git-bot committed:
account_payment_order 15.0.1.0.4

Commit:
[FIX] Restore payment order number in account move line label.

Commit:
[UPD] Update account_payment_order.pot

OCA-git-bot committed:
account_payment_order 15.0.1.0.5

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: bank-payment-15.0/bank-payment-15.0-account_payment_order
Translate-URL: https://translation.odoo-community.org/projects/bank-payment-15-0/bank-payment-15-0-account_payment_order/

wpichler committed:
[IMP] account_payment_order: pre-commit stuff

wpichler committed:
[MIG] account_payment_order: Migration to 16.0

Fixed tests. Removed overwriten _get_default_journal function - it does not exists any more in original object. Removed sepa_in journal in tests - it was never in the manifest as a dependencie - and it does not really belong here

Update account_payment_order/models/account_payment.py

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

Update account_payment_order/models/account_payment.py

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

Update account_payment_order/models/account_payment_order.py

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

Update account_payment_order/wizard/account_payment_line_create.py

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

Update account_payment_order/wizard/account_payment_line_create.py

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

Update account_payment_order/models/account_move.py

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

Update account_payment_order/models/account_move_line.py

account.account internal_type changed to account type with receivable=asset_receivable and payable=liability_payable

Co-authored-by: dzungtran89 <46039081+dzungtran89@users.noreply.github.com>

pedrobaeza committed:
[REF+IMP] account_payment_order: Use native payments

The previous approach creates manually the journal entries and does all
the hard work, plus not being 100% compatible with the bank statement
reconciliation widget (requiring a patch on OCB to see blue lines).

That decision made sense on the moment it was done (v9), where the
native payment model (account.payment) was very limited, and wasn't able
to store all the needed information for the bank transaction.

Now that the limitations are gone, we can get rid off this extra model,
and generate instead `account.payment` records, using both the native
model + methods to perform the same operations.

This serves also to workaround the problem found in #966.

All the code, views and tests of main module have been adapted to this
new approach in this commit. Later commits will adapt the rest of the
modules of the suite, and add migration scripts to transit from the
previous approach to this new one.

TT39832

pedrobaeza committed:
[OU-ADD] account_payment_order: Migration scripts to native payment ref

This adapts the old bank.payment.line records to account.payment records
according the new approach.

dzungtran89 commented on account_payment_order/models/account_move_line.py:
```diff
@@ -0,0 +1,102 @@
+                ml.move_id.move_type in ("in_invoice", "in_refund")
+                and not ml.reconciled
+                and ml.payment_mode_id.payment_order_ok
+                and ml.account_id.internal_type in ("receivable", "payable")
```
```suggestion
                and ml.account_id.account_type in ("asset_receivable", "liability_payable")
```

dzungtran89 commented on account_payment_order/models/account_move.py:
```diff
@@ -0,0 +1,139 @@
+                lambda x: (
+                    not x.reconciled
+                    and x.payment_mode_id.payment_order_ok
+                    and x.account_id.internal_type in ("receivable", "payable")
```
```suggestion
                    and x.account_id.account_type in ("asset_receivable", "liability_payable")
```

dzungtran89 commented on account_payment_order/wizard/account_payment_line_create.py:
```diff
@@ -0,0 +1,177 @@
+            # will not be refunded with a payment.
+            domain += [
+                ("credit", ">", 0),
+                ("account_id.internal_type", "in", ["payable", "receivable"]),
```
```suggestion
                ("account_id.account_type", "in", ["liability_payable", "asset_receivable"]),
```

dzungtran89 commented on account_payment_order/wizard/account_payment_line_create.py:
```diff
@@ -0,0 +1,177 @@
+        elif self.order_id.payment_type == "inbound":
+            domain += [
+                ("debit", ">", 0),
+                ("account_id.internal_type", "in", ["receivable", "payable"]),
```
```suggestion
                ("account_id.account_type", "in", ["asset_receivable", "liability_payable"]),
```

dzungtran89 commented on account_payment_order/models/account_payment_order.py:
```diff
@@ -0,0 +1,616 @@
+                        "paylines": payline,
+                        "total": payline.amount_currency,
+                    }
+            order.recompute()
```
```suggestion
            order.env.flush_all()
```
because it calls a no longer existed method https://github.com/odoo/odoo/blob/16.0/odoo/models.py#L6106

dzungtran89 commented on account_payment_order/models/account_payment.py:
```diff
@@ -0,0 +1,43 @@
+                pay.hide_payment_method_line = (
+                    len(pay.available_payment_method_line_ids) == 1
+                    and pay.available_payment_method_line_ids.code == "manual"
+                )
```
```suggestion
```
Ref: [odoo/odoo#74741](https://github.com/odoo/odoo/pull/74741)

dzungtran89 commented on account_payment_order/models/account_payment.py:
```diff
@@ -0,0 +1,43 @@
+                    pay.payment_type
+                ).filtered(lambda x: not x.payment_method_id.payment_order_only)
+            )
+            to_exclude = self._get_payment_method_codes_to_exclude()
```
```suggestion
            to_exclude = pay._get_payment_method_codes_to_exclude()
```

antony-psc committed:
[FIX] account_payment_order: Error in Batch Payments when deleting a Batch Content line

pedrobaeza commented:
#1000 should be needed, but first is to have green CI. Check also the proper PR title that I have put.

wpichler commented:
#1000 was already included because of the review of [dzungtran89](https://github.com/dzungtran89)

pedrobaeza commented:
And what about the CI?

wpichler commented:
CI in progress now

wpichler commented:
@pedrobaeza done - anything else ?

pedrobaeza commented:
Yes, please check pre-commit changes that are not complete, and please squash all your migration commits into one (except the first one `[IMP] : pre-commit stuff` that is missing the module name by the way.

wpichler commented:
@pedrobaeza please /ocabot merge

fcvalgar commented:
I detected some wrong view in Create payment lines.


https://www.loom.com/share/ca4b41d0e9bf4257aeb38b433982f8e0

wpichler commented:
/ocabot migration account_payment_order

wpichler commented:
> I detected some wrong view in Create payment lines.
> 
> https://www.loom.com/share/ca4b41d0e9bf4257aeb38b433982f8e0

I've corrected the wrong view

![image](https://user-images.githubusercontent.com/13257734/210128147-c63bc79a-9788-4627-83f0-d10306f5d209.png)

pedrobaeza reviewed (changes requested):
This should include #979 before merging, for removing the old stuff. I can do it when I finish the 15 fw-port if you want for not disturbing you more.

wpichler commented:
> 

@pedrobaeza I've looked at #979 - there are many conflicts to resolve. Maybe it is easier if you do continue at this point

wpichler committed:
[FIX] Fixed bug from cherry-pick in the tests

wpichler committed:
[FIX] Make pylint_odoo happy - removed string attribute with same name as the variable

wpichler commented:
> This should include #979 before merging, for removing the old stuff. I can do it when I finish the 15 fw-port if you want for not disturbing you more.

I've now managed to get #979 into this PR - please take a look at it

wpichler committed:
[FIX] Fixed account_internal_type usage

wpichler committed:
[FIX] Fixed bugs seen in the tests

wpichler commented:
@pedrobaeza I would be happy to get some help here - i have problems with the communication in test_supplier_manual_refund. It should be "F1242 R1234" - but is only "F1242". As far as i can see it will get set from the move ref - but the move.ref is F1242. I do not find the point where it should get set to F1241 R1234. It did work before i picked #979. Do you an idea what could cause the problem ?

pedrobaeza commented:
Uhm, weird, as I didn't need to change that. I will check when I have some spare time. First I want to do the fw-port to v15.

wpichler commented:
Found it - this was missing: https://github.com/OCA/bank-payment/pull/919

wpichler committed:
[IMP] Pre Commit Changes

wpichler commented:
Rebased - all checks are fine - requested changes are done. Please review and approve

ivantodorovich commented on account_payment_order/migrations/14.0.1.3.0/pre-migration.py:
```diff
@@ -0,0 +1,10 @@
+    if not version:
+        return
+
+    cr.execute("UPDATE account_payment_order SET state='uploaded' WHERE state='done'")
```
This file should be removed

(TO ANALYZE: or moved to a 16.0.1.0.0 migration --> and if so, it must not fail if this refactor has already been migrated in previous versions)

pedrobaeza replied:
Yes, to remove. Not needed in v16, although the v15 step is still pending.

ivantodorovich commented on account_payment_order/migrations/14.0.2.0.0/post-migration.py:
```diff
@@ -0,0 +1,174 @@
+        env.cr, "ALTER TABLE account_payment ALTER move_id SET NOT NULL"
+    )
+    _delete_hooks(env)
+    _insert_payment_line_payment_link(env)
```
This file should be removed, too

(TO ANALYZE: or moved to a 16.0.1.0.0 migration --> and if so, it must not fail if this refactor has already been migrated in previous versions)

ivantodorovich commented on account_payment_order/report/account_payment_order.xml:
```diff
@@ -0,0 +1,158 @@
+                <div class="oe_structure" />
+                <div class="row">
+                    <div class="col-4 offset-8">
+                        <span
+                            t-raw="'%s &lt;br&gt;' % doc.journal_id.bank_id.name if doc.journal_id.bank_id.name else ''"
+                        />
```
Using `t-raw` here doesn't seem like a good idea here.
IMO, if we want each field in a new line, we should use `div`.
Using `t-field` will automatically hide it if the field is empty, so it's also nicer

```suggestion
                        <div t-field="doc.journal_id.bank_id.name" />
```

wpichler replied:
This is also resolved

hildickethan commented on account_payment_order/views/account_payment_line.xml:
```diff
@@ -9,6 +9,7 @@
                     <group name="left">
                         <field
                             name="order_id"
+                            groups="base.group_multi_company"
                             invisible="not context.get('account_payment_line_main_view')"
                         />
                         <field name="name" />
```
The proper way to fix this is adding an invisible `company_id` field that doesnt have the `groups="base.group_multi_company"` restriction. The comment you wrote in the PR about this error is to my knowledge some new behaviour in 16.0, but the tl;dr is that the `company_id` field in this view needs the multi-company group so it depends on that in order to use the `company_id` domain it adds automatically with `check_company`.

If you add `<field name="company_id" invisible="1" />` above it, it will work without this groups restriction, which is better since something like an order field shouldn't really be restricted to multi company

wpichler replied:
Thanks - this is done

hildickethan commented:
[[FIX] Fixed bug from cherry-pick in the tests](https://github.com/OCA/bank-payment/pull/972/commits/52e1948d1b67ee867650efd822c9c6e03b9b34cf)
[[FIX] Make pylint_odoo happy - removed string attribute with same nam…](https://github.com/OCA/bank-payment/pull/972/commits/8d659116694d8d053c861647592630422d99f179) 
[[FIX] Fixed account_internal_type usage](https://github.com/OCA/bank-payment/pull/972/commits/eae350748c09011993e1e2e0703fad78a31fc98a)
[[FIX] Fixed bugs seen in the tests](https://github.com/OCA/bank-payment/pull/972/commits/93cbc872066d4fd0b94a8205847eefc39069098e)

these commits should be squashed into the migration commit

[[IMP] Pre Commit Changes](https://github.com/OCA/bank-payment/pull/972/commits/9b437af4bdb9787ef8b97c59cfb62f48dcf56071)

should be squashed into the pre-commit stuff before the migration or if they are just fixing your migrated changes, then they should be part of the original migration too

i would also put the cherry-picked ones before the migration commit, in case anything has to be adapted to 16.0

astirpe commented on account_payment_order/models/account_payment_order.py:
```diff
@@ -0,0 +1,444 @@
+            order.move_count = mapped_data.get(order.id, 0)
+
+    @api.model
+    def create(self, vals):
```
I just tried to install the module locally and I notice the following warnings:

```
2023-02-08 14:15:52,874 1088979 WARNING db123 odoo.api.create: The model odoo.addons.account_payment_order.models.account_payment_order is not overriding the create method in batch 
2023-02-08 14:15:52,876 1088979 WARNING db123 odoo.api.create: The model odoo.addons.account_payment_order.models.account_payment_line is not overriding the create method in batch 

```

Could you change the create() method to support the creation in batch and make use of `@api.model_create_multi` ?

wpichler replied:
This got corrected by [AnizR](https://github.com/AnizR) - merge PR into this PR

AnizR committed:
[IMP] create with @api.model_create_multi tag
[MIG] migration scripts
[IMP] report account_payment_order use t-field
[IMP] field company_id invisible on view account_payment_line

wpichler committed:
Merge pull request #1 from acsone/16.0-mig-account_payment_order

[IMP] create with @api.model_create_multi tag

wpichler commented:
i've merge PR from @AnizR and pick squashed all commits into one migration commit. I hope this is the correct way do keep history of commits as it should be

pedrobaeza commented:
I'm afraid you have squashed too much. Commits like the account.payment generation should be isolated, or the extra fixes. Can you please undo that?

wpichler commented:
@pedrobaeza I've undone it - and have no idea how to squash these commits - any hint ?

wpichler committed:
[TRA] Added Better Translations for de_DE

AntoniRomera commented:
Need any help?

wpichler commented:
> Need any help?

Yes please

AntoniRomera commented:
Show a picture of the commit history to see how you currently have the branch.

mmequignon commented on account_payment_order/migrations/16.0.1.0.0/post-migration.py:
```diff
@@ -161,6 +164,18 @@ def _insert_payment_line_payment_link(env):
 
 @openupgrade.migrate()
 def migrate(env, version):
+    env.cr.execute(
```
Do we want to execute this when this has already been executed during `14.0.2.0.0` migration?
This looks valid only if migrating from `14.0.1.0.0` to `16.0.1.0.0`.
Same applies to `14.0.3.0.0`

pedrobaeza commented:
I have put a bit of order + gather latest changes from previous branches on #1056, so closing this one and merging the other as soon as I get a green CI. Thanks all for your efforts.
